src/engine/cachingreader/cachingreader.h

```diff
@@ -111,7 +111,7 @@ class CachingReader : public QObject {
   signals:
     // Emitted once a new track is loaded and ready to be read from.
     void trackLoading();
-    void trackLoaded(TrackPointer pTrack, int iSampleRate, int iNumSamples);
+    void trackLoaded(TrackPointer pTrack, int trackSampleRate, double trackNumSamples);
```

Comment:
why do you prefer double of `int64_t`/`long long int` in this case?

Reply:
This is used as double in the receiver EngineBuffer::slotTrackLoaded() and was double in CachingReaderWorker::loadTrack() the change is a fix.

Reply:
yes but why use `double` when we only want/expect precise integer values?

Reply:
At one point we need to go from the Audio Source SINT to double. The original 2.3 code does it in the CachingReaderWorker. Using int for passing this value was the bug. double Int double.
Now it is double double double.

Reply:
Right that makes sense, now why is the engine using `double`?

Reply:
The engine transport is controlled by the output after resampling in whole frames. Fractional rates are causing fractional input positions from the sound source. That's why the engine buffer is using double for positions.

Reply:
right, but that's for positions, why do we have fractional lengths? There are no sample buffers with fractional sizes, right?

Reply:
That is right. The lengths is also the end position. Sure, we can change that to SINT and compare it to the double position on the fly. But that will be a refactoring PR and not a bug fix. I also don't see a benefit to change that.

Reply:
Well if we changed that to SINT, wouldn't we suffer the overflow issue again? The thing I'm worried about is that the code inside those methods will get more difficult since now all of a sudden we might get fractional lengths passed. That will make the code more difficult to reason about. I think a comment and/or a DEBUG_ASSERT is the least we can do document that thats not the case.